scalableminds/webknossos
# Merge "Shared Annotations" with "My annotations"

- [X] @fm3 Could you have a look at my backend changes? I'm not sure whether this is the right approach (the "compact writes" got a bit more expensive due to the additional look ups). Also, I'm not sure how the pagination should be built when merging "own" with "shared" annotations.

### URL of deployed dev instance (used for testing):
- https://dashboardcollab.webknossos.xyz

### Steps to test:
- Set up two users (one should already exist; for the second one, register in an incognito browser tab and enable the account from the first user)
- Create multiple annotations from both accounts
- Assign a common team to both users and share some annotations with that team
- The dashboard "annotation" tab should always show the own annotations *and* the ones which are shared with that user
- Shared annotations should not be editable via the "Archive", "Archive All" or "Edit Name" buttons.
- Test the filtering UI in the "Owners & Teams" column

### Todo:
- [x] unify `user` and `owner` attribute in annotations

### Issues:
- fixes #6142

------
(Please delete unneeded items, merge only when none are left open)
